flaky TestPing

```
  === RUN   TestPing
  2022/06/24 14:44:29 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
      ping_test.go:47: ping took:  43.4µs
      ping_test.go:47: ping took:  8.205732ms
      ping_test.go:47: ping took:  107.001µs
      ping_test.go:47: ping took:  77.8µs
      ping_test.go:47: ping took:  66.3µs
      ping_test.go:46: 
          	Error Trace:	ping_test.go:46
          	            				ping_test.go:35
          	Error:      	Received unexpected error:
          	            	Write on stream 5 canceled with error code 0
          	Test:       	TestPing
  --- FAIL: TestPing (0.10s)

--- a/p2p/protocol/ping/ping_test.go
+++ b/p2p/protocol/ping/ping_test.go
@@ -5,13 +5,14 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
-
 	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
 	swarmt "github.com/libp2p/go-libp2p/p2p/net/swarm/testing"
 	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
 
 	"github.com/libp2p/go-libp2p-core/peer"
+
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
 )
 
 func TestPing(t *testing.T) {
@@ -19,12 +20,14 @@ func TestPing(t *testing.T) {
 	defer cancel()
 	h1, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
 	require.NoError(t, err)
+	defer h1.Close()
 	h2, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
 	require.NoError(t, err)
+	defer h2.Close()
 
 	err = h1.Connect(ctx, peer.AddrInfo{
 		ID:    h2.ID(),
-		Addrs: h2.Addrs(),
+		Addrs: []ma.Multiaddr{h2.Addrs()[0]},
 	})
 	require.NoError(t, err)
 

